### PR TITLE
Improve help with command summary table

### LINKED ISSUE #29: As user who is not too familiar with the commands, I can to have command guides handy at any point of time so that I can easily refer to them.

### PR DESCRIPTION
**Pull request by Xiaoyunnn**
Fixes #29.

TODO:
- [x] Replace the getters for format and examples with static variables after the quick start branch has been merged.

When user types `help`, help window appears:
image

When the user double clicks format/example cell, the cell becomes editable to allow the text content to be selected and copied:
image
_For some unknown reasons, double-clicking to make the cells selectable is not responsive sometimes.*_

The user can enter any nonsense when the cell is editable, but when the user presses `Enter` or clicks other areas, the content changes back to the pre-defined text (i.e. the correct formats and examples). After the user clicks `Enter`, the cell becomes like this*:
image
_*I think these are limitations of JavaFX rather than bugs.🤔_

But it will be restored back to the wrapped-text form when the user clicks elsewhere on the table.

The user can sort the rows alphabetically using by clicking the right end of each column header:
image

Double-clicking the column end allows the column to expand into a single line (may look better for shorter commands but a lot of our commands are very long which makes it less convenient to scroll through when the user expands the columns):
image

### CODE REVIEW

**Comment by Chesterwongz** [comment]
Perhaps you could make this PR a draft PR until the TODO things and bug fixes have been added, so that the PR space wont be too cluttered, and the others will know when it is ready for review

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -32,12 +34,42 @@
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
 
-    private final Index targetIndex;
+    private Index targetIndex;
```
[the remark this answers is not in the document]
Comment (eeliana) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:37]: Okay I understand the reasoning for implementing it this way, but personal preference is I'm not really that comfortable with the idea of having public empty constructors. Do you think the ObservableList<Command> in HelpWindow can be placed in util as SampleCommandUtil instead (similar to SampleDataUtil) or is that unnecessary? I'm not really sure also..

[the remark this answers is not in the document]
Comment (Xiaoyunnn) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:37]: I initially had a `CommandSummary` class with `action`, `format`, and `example` as private fields and then followed by the getters. But then I felt it is repetitive so I removed it and added the indiv getters in the command subclasses. I'm not sure which is a better representation also... 😶

**Comment by Xiaoyunnn** [comment]
Changed copy cell values method.

To copy cell value, select the table cell (background color changes to a darker purple): 
image

Press CTRL + C or CMD + C to copy the cell value to the system clipboard.

To copy multiple cell values, press SHIFT + MOUSE CLICK or SHIFT + keyboard up down left right and followed by CTRL + C or CMD + C:
image

I tried making the help window resizable but cannot find a suitable method to get the columns to resize automatically🥲 (there will be an empty column when I resize the window horizontally). So I'm planning to keep it not resizable;-;

File: src/main/resources/view/HelpWindow.fxml [hunk 2]
```diff
@@ -19,27 +23,48 @@
...
+                           <Insets bottom="7.0" left="12.0" right="12.0" top="7.0" />
+                        </padding></Button>
+              </children>
+              <opaqueInsets>
+                <Insets bottom="10.0" left="5.0" right="10.0" top="5.0" />
+              </opaqueInsets>
+              <padding>
+                <Insets bottom="10.0" left="5.0" right="10.0" top="5.0" />
+              </padding>
+            </HBox>
```
Comment (eeliana) [inline src/main/resources/view/HelpWindow.fxml:65]: image
Is there a way to make this inherit the size of parent after resizing the help window?

**Review by eeliana — APPROVED** [review]
LGTM

File: src/main/resources/view/HelpWindow.fxml [hunk 3]
```diff
@@ -19,27 +23,48 @@
...
+                  </opaqueInsets>
+               </TableView>
+            <HBox alignment="CENTER" layoutX="1.0" layoutY="536.0" minHeight="80.0" minWidth="550.0" style="-fx-background-color: D3DAFB50;">
+              <children>
+                <Label fx:id="helpMessage" text="Label" textFill="#424874" wrapText="true" HBox.hgrow="ALWAYS">
+                    <padding>
+                       <Insets bottom="5.0" left="10.0" right="10.0" top="5.0" />
+                    </padding>
+                </Label>
+                <Button fx:id="copyButton" mnemonicParsing="false" onAction="#copyUrl" text="Copy URL" textFill="WHITE" wrapText="true">
```
Comment (lingshanng) [inline src/main/resources/view/HelpWindow.fxml:45]: is the copy url button suppose to change size when hover over? maybe it'll be better if it doesn't bcos it makes the text move. But then after clicking on a table cell, it doesn't change size on hovering until its clicked on...

File: src/main/resources/view/HelpWindow.fxml [hunk 4]
```diff
@@ -19,27 +23,48 @@
...
+                   <TableColumn fx:id="format" prefWidth="232.0" reorderable="false" text="Format" />
+                     <TableColumn fx:id="example" prefWidth="229.0" reorderable="false" text="Example" />
+                 </columns>
+                  <opaqueInsets>
+                     <Insets />
+                  </opaqueInsets>
+               </TableView>
+            <HBox alignment="CENTER" layoutX="1.0" layoutY="536.0" minHeight="80.0" minWidth="550.0" style="-fx-background-color: D3DAFB50;">
+              <children>
+                <Label fx:id="helpMessage" text="Label" textFill="#424874" wrapText="true" HBox.hgrow="ALWAYS">
```
Comment (lingshanng) [inline src/main/resources/view/HelpWindow.fxml:40]: not so important but the padding missing when minimised
image

Also just my opinion but the footer can have different color to differentiate from the table.
The scroll bar a bit hard to see as well.

File: src/main/resources/view/HelpWindow.fxml [hunk 5]
```diff
@@ -5,11 +5,15 @@
...
+<?import javafx.scene.control.TableColumn?>
+<?import javafx.scene.control.TableView?>
 <?import javafx.scene.image.Image?>
 <?import javafx.scene.layout.HBox?>
+<?import javafx.scene.layout.VBox?>
+<?import javafx.scene.text.Font?>
 <?import javafx.stage.Stage?>
 
-<fx:root resizable="false" title="Help" type="javafx.stage.Stage" xmlns="http://javafx.com/javafx/11" xmlns:fx="http://javafx.com/fxml/1">
+<fx:root minHeight="600.0" minWidth="550.0" title="Help" type="javafx.stage.Stage" xmlns="http://javafx.com/javafx/16" xmlns:fx="http://javafx.com/fxml/1">
```
Comment (lingshanng) [inline src/main/resources/view/HelpWindow.fxml:16]: Can make the help window a bit bigger? The add command a bit long

Comment (lingshanng) [reply src/main/resources/view/HelpWindow.fxml:16]: Also can make the font size bigger?

File: src/main/resources/view/HelpWindow.fxml (hunk 4 above)
Comment (eeliana) [reply src/main/resources/view/HelpWindow.fxml:40]: Padding issue not observed on my side as well. I prefer a more visible scroll bar as well but not a major issue.

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 6]
```diff
@@ -2,46 +2,118 @@
...
+    public static final String HELP_MESSAGE = "Select the table cell(s) and press CTRL + C or CMD + C to copy.\n"
+            + "Refer to the user guide:\n" + USERGUIDE_URL;
 
     private static final Logger logger = LogsCenter.getLogger(HelpWindow.class);
     private static final String FXML = "HelpWindow.fxml";
 
+    private final ObservableList<CommandSummary> list = CommandSummary.getCommandSummaryList();
+
+    @FXML
+    private TableView<CommandSummary> table;
```
[the remark this answers is not in the document]
Comment (lingshanng) [reply src/main/java/seedu/address/ui/HelpWindow.java:41]: oh i see, in that case i think can just leave it, not v important

**Pull request merged**